src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -224,12 +227,18 @@ void ucc_tl_ucp_reduce_scatter_knomial_progress(ucc_coll_task_t *coll_task)
     void                    *local_data;
     int                      is_avg;
 
+    if (args->coll_type == UCC_COLL_TYPE_REDUCE) {
+        root = args->root;
+        rank = VRANK(rank, root, size);
+    }
+
     UCC_KN_REDUCE_GOTO_PHASE(task->reduce_scatter_kn.phase);
     block_count = ucc_sra_kn_compute_block_count(count, rank, p);
     get_rs_work_buf(task, block_count, &wb);
     if (KN_NODE_EXTRA == node_type) {
         peer = ucc_ep_map_eval(task->subset.map,
```

Comment:
I see that in many places we are trying to find out the actual peer. Can you add a function call that unifies all these ```VRANK/INV_VRANK/ucc_ep_map_eval``` calls into a ```knomial_get_actual_peer()``` and use this function instead?

Reply:
done